front/src/reducers/user/index.ts

```diff
@@ -1,118 +1,44 @@
-import { createSlice, Dispatch, PayloadAction } from '@reduxjs/toolkit';
-import jwtDecode from 'jwt-decode';
-
-import keycloak from 'keycloak';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { ApiError } from 'common/api/baseEditoastApi';
 
 export interface UserState {
   isLogged: boolean;
-  toLogin: boolean;
-  loginError: boolean;
-  serverError: boolean;
-  username: string;
-  accessToken?: string;
+  loginError?: ApiError;
+  username?: string;
```

Comment:
Username cannot be undefined since by default it is an empty string
```suggestion
  username: string;
```